instrumentation/opentelemetry-api-beta/src/test/groovy/ContextBridgeTest.groovy

```diff
@@ -0,0 +1,149 @@
+        agentContext2.makeCurrent().withCloseable {
+          io.opentelemetry.context.Context.current().get(FOOD) == "cheese"
+          io.opentelemetry.context.Context.current().get(COUNTRY) == "japan"
+          Context.current().get(ANIMAL) == "cat"
+        }
+      }
+    }
+  }
+
+  // The difference between "standard" context interop and our bridge is that with normal interop,
+  // keys are still isolated completely. We have special logic to share the same data for our known
+  // types like span.
+  def "agent and application share span"() {
+    when:
+    def applicationTracer = OpenTelemetry.getGlobalTracer("test")
+    def agentTracer = io.opentelemetry.OpenTelemetry.getGlobalTracer("test")
+
+    then:
+    !Span.current().spanContext.isValid()
+    !io.opentelemetry.trace.Span.current().spanContext.isValid()
+
+    def applicationSpan = applicationTracer.spanBuilder("test1").startSpan()
+    applicationSpan.spanContext.isValid()
+    applicationSpan.makeCurrent().withCloseable {
+      Span.current().spanContext.spanIdAsHexString == applicationSpan.spanContext.spanIdAsHexString
+      io.opentelemetry.trace.Span.current().spanContext.spanIdAsHexString == applicationSpan.spanContext.spanIdAsHexString
+
+      def agentSpan = agentTracer.spanBuilder("test2").startSpan()
+      agentSpan.makeCurrent().withCloseable {
+        Span.current().spanContext.spanIdAsHexString == agentSpan.spanContext.spanIdAsHexString
+        Span.current().spanContext.traceIdAsHexString == agentSpan.spanContext.spanIdAsHexString
+        Span.current().spanContext.traceIdAsHexString == applicationSpan.spanContext.spanIdAsHexString
+        io.opentelemetry.trace.Span.current().spanContext.spanIdAsHexString == agentSpan.spanContext.spanIdAsHexString
+        io.opentelemetry.trace.Span.current().spanContext.traceIdAsHexString == agentSpan.spanContext.traceIdAsHexString
+        io.opentelemetry.trace.Span.current().spanContext.traceIdAsHexString == applicationSpan.spanContext.traceIdAsHexString
+
+        def applicationSpan2 = applicationTracer.spanBuilder("test3").startSpan()
+        applicationSpan2.makeCurrent().withCloseable {
+          Span.current().spanContext.spanIdAsHexString == applicationSpan2.spanContext.spanIdAsHexString
+          Span.current().spanContext.traceIdAsHexString == applicationSpan2.spanContext.spanIdAsHexString
+          Span.current().spanContext.traceIdAsHexString == applicationSpan.spanContext.spanIdAsHexString
+          io.opentelemetry.trace.Span.current().spanContext.spanIdAsHexString == applicationSpan2.spanContext.spanIdAsHexString
+          io.opentelemetry.trace.Span.current().spanContext.traceIdAsHexString == applicationSpan2.spanContext.traceIdAsHexString
+          io.opentelemetry.trace.Span.current().spanContext.traceIdAsHexString == applicationSpan.spanContext.traceIdAsHexString
+        }
+      }
+    }
+  }
+
+  def "agent wraps"() {
```

Comment:
these tests run with our java-concurrent, so I think that will auto-propagate agentContext(?) which makes it hard to reason about this test

maybe cleaner to do the `Context.root().makeCurrent().withScoped {}` on both, then call `context.wrap(runnable).run()` inside of that, and not mess with threads?

Reply:
Heh forgot about that. Yeah that seems like it will test it well.